### PR TITLE
serve local file for background image

### PR DESCRIPTION
**Pull request by eagleoflqj**
Please update submodule on merge.

### CODE REVIEW

File: src/config/imageoptionview.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+private let imageDir = wwwDir.appendingPathComponent("img")
+
+struct ImageOptionView: OptionView {
+  let label: String
+  let overrideLabel: String? = nil
+  @ObservedObject var model: ImageOption
+  @State private var openPanel = NSOpenPanel()
+
+  func selectImage() {
+    mkdirP(imageDir.localPath())
```
Comment (ksqsf) [inline src/config/imageoptionview.swift:18]: (This seems not satisfying, but acceptable anyway)

File: src/config/imageoptionview.swift [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    openPanel.directoryURL = imageDir
+    openPanel.begin { response in
+      if response == .OK {
+        guard let file = openPanel.urls.first else {
+          return
+        }
+        var fileName = file.lastPathComponent
+        if !imageDir.contains(file) {
+          if !copyFile(file, imageDir.appendingPathComponent(fileName)) {
+            return
```
Comment (ksqsf) [inline src/config/imageoptionview.swift:32]: I think we should add an error message here, at least in the log file

**Review by ksqsf — APPROVED** [review]
LGTM

File: src/config/imageoptionview.swift (hunk 2 above)
Comment (eagleoflqj) [reply src/config/imageoptionview.swift:32]: copyFile already [logs](https://github.com/fcitx-contrib/fcitx5-macos/blob/e14a44ae651c99b258497e17a8998dfb67558495/src/config/util.swift#L78) on fail.
What I want to improve in the future is to provide a convenient way to trigger an AlertToast from nested subview to window's major view. Currently the only place that does the similar thing is https://github.com/fcitx-contrib/fcitx5-macos/blob/e14a44ae651c99b258497e17a8998dfb67558495/src/config/inputmethod.swift#L215-L228 but I'm too weak Swift programmer to make it flexible 🌚

**Pull request merged**